Il2CppInterop.Common/XrefScans/XrefScanUtilFinder.cs

```diff
@@ -94,4 +94,10 @@ private static ulong ExtractTargetAddress(in Instruction instruction)
                 return 0;
         }
     }
+
+    private static IntPtr ConvertUlongToIntPtr(ulong address)
+    {
+        if (Environment.Is64BitProcess) return address <= long.MaxValue ? new IntPtr((long)address) : new IntPtr(unchecked((long)address));
+        else return address <= int.MaxValue ? new IntPtr((int)address) : new IntPtr(unchecked((int)address));
```

Comment:
This is just `return unchecked((IntPtr)address);`

However, dropping the upper 32 bits seems like a massive bug. Did you actually verify that this works?

Reply:
I'm extremely hesitant to merge this kind of change. I want a review by one of the other maintainers.